### ISSUE TITLE
Migrate to Microsoft.Extensions.DependencyInjection

### ISSUE THREAD

**Issue by gickis** [noobot/noobot#99]
I think it's good idea to migrate from concrete DI. So as result end user will decide what DI container to use in real app.

**Comment by flavourous** [comment]
I agree - I want to put this in an Xamarin.iOS app, but structuremap won't run on iOS (looks like it uses emit?).  Is Microsoft.Extensions.DependencyInjection in .net standard?  It would be a shame to use something that's only in asp.netcore.

**Comment by AshleyPoole** [comment]
I have a forked version of the framework that now allows the consumer to decide what DI container to use called CBot - https://github.com/AshleyPoole/CBot

Here's an example usage as well - https://github.com/AshleyPoole/SreBot
